server/services/caCaseloadService.ts

```diff
@@ -0,0 +1,440 @@
+          }),
+      }
+    })
+  }
+
+  private async mapResponsibleComsToCasesWithExclusions(
+    caseload: Container<ManagedCase>
+  ): Promise<Container<ManagedCase>> {
+    const comUsernames = caseload
+      .unwrap()
+      .map(
+        offender =>
+          offender.licences.find(l => offender.licences.length === 1 || l.status !== LicenceStatus.ACTIVE).comUsername
+      )
+      .filter(comUsername => comUsername)
+
+    const coms = await this.communityService.getStaffDetailsByUsernameList(comUsernames)
+
+    return caseload.map(offender => {
+      const responsibleCom = coms.find(
+        com =>
+          com.username?.toLowerCase() ===
+          offender.licences
+            .find(l => offender.licences.length === 1 || l.status !== LicenceStatus.ACTIVE)
+            .comUsername?.toLowerCase()
+      )
+
+      if (responsibleCom) {
+        return {
+          ...offender,
+          probationPractitioner: {
+            staffCode: responsibleCom.staffCode,
+            name: `${responsibleCom.staff.forenames} ${responsibleCom.staff.surname}`.trim(),
+          },
+        }
+      }
+
+      if (!offender.deliusRecord.staff || offender.deliusRecord.staff.unallocated) {
+        return {
+          ...offender,
+        }
+      }
+
+      return {
+        ...offender,
+        probationPractitioner: {
+          staffCode: offender.deliusRecord.staff.code,
+          name: `${offender.deliusRecord.staff.forenames} ${offender.deliusRecord.staff.surname}`.trim(),
+        },
+      }
+    })
+  }
+
+  private isRecall = (offender: ManagedCase): boolean => {
+    const recall = offender.nomisRecord?.recall && offender.nomisRecord.recall === true
+    const crd = offender.nomisRecord?.conditionalReleaseDate
+    const prrd = offender.nomisRecord?.postRecallReleaseDate
+
+    // If a CRD but no PRRD it should NOT be treated as a recall
+    if (crd && !prrd) {
+      return false
+    }
+
+    if (crd && prrd) {
+      const dateCrd = moment(offender.nomisRecord.conditionalReleaseDate, 'YYYY-MM-DD')
+      const datePrrd = moment(offender.nomisRecord.postRecallReleaseDate, 'YYYY-MM-DD')
+      // If the PRRD > CRD - it should be treated as a recall
+      if (datePrrd.isAfter(dateCrd)) {
+        return true
+      }
+      // If PRRD <= CRD - should not be treated as a recall
+      return false
+    }
+
+    // Trust the Nomis recall flag as a fallback position - the above rules should always override
+    return recall
+  }
+
+  private isBreachOfTopUpSupervision = (offender: ManagedCase): boolean => {
+    return offender.nomisRecord?.imprisonmentStatus && offender.nomisRecord?.imprisonmentStatus === 'BOTUS'
+  }
+
+  /**
+   * Parole Eligibility Date must be set and in the future
+   * If the date is in the past, it's no longer parole eligible
+   * Parole eligibility excludes the offender, so a truthy return here is an exclusion from CVL
+   * @param ped
+   */
+  public static isParoleEligible(ped: string): boolean {
```

Comment:
all of this is copied across as is and then trimmed back a bit. 
Going to extract out some of these static helpers into somewhere central to share across the caseload builders - these can be reimplemented in the backend